go-controller/pkg/config/utils.go

```diff
@@ -16,6 +16,18 @@ type HostPort struct {
 	Port int32
 }
 
+// String representation of a HostPort entry
+func (hp *HostPort) String() string {
+	switch {
+	case hp.Host == nil:
+		return fmt.Sprintf(":%d", hp.Port)
+	case len(*hp.Host) == net.IPv6len:
```

Comment:
The canonical way of doing this is 
```go
if hp.Host.To4() != nil {
    // ... is ipv4
} else {
  // .. is ipv6
}
```

Reply:
You are right... I was digging into the `To4()` code I misunderstood it.